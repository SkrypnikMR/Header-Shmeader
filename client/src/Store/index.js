import { combineReducers } from 'redux';
import { reducer as chatReducer } from './chat/reducer';
import { reducer as registrationReducer } from './registration/reducer';
import { reducer as loginReducer } from './login/reducer';
import { reducer as userReducer } from './user/reducer';

const rootReducer = combineReducers({
  chat: chatReducer,
  registration: registrationReducer,
  login: loginReducer,
  user: userReducer,
});


export default rootReducer;
