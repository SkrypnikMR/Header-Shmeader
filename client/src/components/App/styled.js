import styled from 'styled-components';

export const StAppDiv = styled.body`
  height: 100vh;
  ${({ url = 'heh' }) => url && `background: url(${url}) 100% 100% no-repeat`};
  background-size: cover;
`;
