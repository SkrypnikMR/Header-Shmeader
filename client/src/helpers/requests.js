export const getRequest = async (url, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = { method: 'GET', headers };
  const answer = await fetch(url, options);
  return answer.json();
};
export const postRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'POST',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(url, options);
  return answer.json();
};

export const putRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'PUT',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(url, options);
  return answer.json();
};
export const createAuthHeader = (support) => { return { Authorization: support.getCookie('token') }; };
