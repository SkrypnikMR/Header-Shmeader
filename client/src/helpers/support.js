export const support = {
    setSessionStorageItem: (name, data) => {
        if (typeof data !== 'string') data = JSON.stringify(data);
        sessionStorage.setItem(name, data);
    },
    killSessionStorageItem: (name) => {
        sessionStorage.removeItem(name);
    },
    getSessionStorageItem: (name) => {
        const item = sessionStorage.getItem(name);
        return item?.charAt(0) === '{' ? JSON.parse(item) : item;
    },
    getPrettyTime: time => `${new Date(Number(time)).toTimeString().substr(0, 8)}`,
    getMessagesFolders: (rooms) => {
        const messagesFolders = {};
        rooms.forEach((room) => {
            const { room_name } = room;
            messagesFolders[room_name] = [];
        });
        return messagesFolders;
    },
    filteredRooms: (filterValue, rooms) => {
        const result = rooms.filter(room => room.room_name.includes(filterValue));
        return result.length === 0 ? null : result;
    },
};
