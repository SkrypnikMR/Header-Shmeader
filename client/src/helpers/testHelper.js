import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

export const shallowSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    const test = <Router>{core}</Router>;
    return shallow(test);
};
