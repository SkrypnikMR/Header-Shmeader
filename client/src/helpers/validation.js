import { regExp } from '../constants/regExp';

export const validation = {
    registrationValidation: ({ email, password, confirm, firstName, lastName }) => {
        let answer = { message: '', isValid: true };

        if (!regExp.emailRegExp.test(email) || !email) answer = { message: 'Invalid email', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) answer = { message: 'Invalid password', isValid: false };
        if (!regExp.firstNameRegExp.test(firstName) || !firstName) answer = { message: 'Invalid firstName', isValid: false };
        if (!regExp.lastNameRegExp.test(lastName) || !lastName) answer = { message: 'Invalid lastName', isValid: false };
        if (password !== confirm) answer = { message: 'Password mismatch', isValid: false };
        return answer;

    },
    loginValidation: ({ email, password }) => {
        let answer = { message: '', isValid: true };

        if (!regExp.emailRegExp.test(email) || !email) answer = { message: 'Invalid email', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) answer = { message: 'Invalid password', isValid: false };
        return answer;
    },
};
