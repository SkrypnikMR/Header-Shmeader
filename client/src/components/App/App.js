import React from 'react';
import { Route, Switch, BrowserRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import Registration from '../Registration';
import Login from '../Login';
import { APP_ROUTES } from '/src/contsants/reactRoutes';
import { StAppDiv } from './styled';


const App = ({ userTheme, userThemeMode }) => {
  return (
    <StAppDiv url={userThemeMode === 'dark' ? userTheme.dark : userTheme.light}>
      <BrowserRouter>
        <Switch>
          <Route path={APP_ROUTES.login} exact component={Login} />
          <Route path={APP_ROUTES.registration} exact component={Registration} />
        </Switch>
      </BrowserRouter>
    </StAppDiv >
  );
};

App.propTypes = {
  userTheme: PropTypes.object,
  userThemeMode: PropTypes.string,
};

export default App;
