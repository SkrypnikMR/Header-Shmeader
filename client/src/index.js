import '../public/styles/styles.scss';
import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from '@redux-saga/core';
import rootReducer from './Store';
import App from './components/App';
import rootSaga from './Store/saga';
import './i18n';

/* const devTools = window?.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(); devTools, */
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(saga),
  ),
);
saga.run(rootSaga);

render(
  <Provider store={store}>
    <App />
  </Provider >,
  document.getElementById('root'),
);
