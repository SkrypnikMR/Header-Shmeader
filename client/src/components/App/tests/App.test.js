import React from 'react';
import configureStore from 'redux-mock-store';
import { shallowSmart } from '/src/helpers/testHelper';
import App from '../App';
import '/src/i18n';

const mockStore = configureStore();
const store = mockStore({
  registration: {
    email: '',
    password: '',
    confirm: '',
    firstName: '',
    lastName: '',
    success: null,
  },
  login: {
    email: '',
    password: '',
    confirm: '',
    firstName: '',
    lastName: '',
    success: null,
  },
  user: {
    themeMode: 'light',
  },
});
const props = {
  userTheme: {
    light: 'lightBackground',
    dark: 'darkBackground',
  },
  userThemeMode: 'dark',
};

describe('App', () => {
  it('Should match snapshot', () => {
    const component = shallowSmart(<App {...props} />, store);
    expect(component.html()).toMatchSnapshot();
  });
});
