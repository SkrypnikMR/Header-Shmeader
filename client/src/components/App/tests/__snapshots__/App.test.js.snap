// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapshot 1`] = `"<body class=\\"styled__StAppDiv-sc-7r44pg-0 jLSCAE\\"><div class=\\"styled__StHeader-zzrvsn-0 hDaupM\\"><div class=\\"styled__StLogo-sc-9vy3sh-0 krciQz\\"><img src=\\"./public/assets/images/logo.png\\"/><h1>Header&amp;Shmeader</h1></div><div class=\\"styled__StControl-sc-1i99xqk-0 joBAxQ\\"><button id=\\"en\\" type=\\"button\\" value=\\"en\\" color=\\"white\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 iwKgks\\">EN</button><button id=\\"ru\\" type=\\"button\\" value=\\"ru\\" color=\\"white\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 iwKgks\\">RU</button><button id=\\"ar\\" type=\\"button\\" value=\\"ar\\" color=\\"white\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 iwKgks\\">AR</button></div></div><div class=\\"styled__StLoginDiv-ohf2t6-0 cCvQFI\\"><div class=\\"styled__StFormDiv-sc-1xa63f-0 yOOBF\\"><p>Sign In</p><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-w3ykst-3 emTECj\\"><label for=\\"email\\" class=\\"styled__StLabel-w3ykst-0 gOEYZv\\">Email</label><input type=\\"text\\" id=\\"email\\" name=\\"email\\" value=\\"\\" placeholder=\\"Enter your email...\\" class=\\"styled__StInput-w3ykst-1 eAZEdV\\"/></div><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-w3ykst-3 emTECj\\"><label for=\\"password\\" class=\\"styled__StLabel-w3ykst-0 gOEYZv\\">Password</label><input type=\\"text\\" id=\\"password\\" name=\\"password\\" value=\\"\\" placeholder=\\"Enter your password...\\" class=\\"styled__StInput-w3ykst-1 eAZEdV\\"/></div><button id=\\"login\\" type=\\"button\\" height=\\"50px\\" class=\\"styled__StButton-sc-1a6hck8-0 biUGaX\\">Login</button><span>Need an account?<a href=\\"/registration\\">Register now</a></span></div></div><div class=\\"notification-container notification-container-empty\\"><div></div></div></body>"`;
